### DISCUSSION TITLE
Configuring region in S3 image storage

### DISCUSSION THREAD

**Discussion by tk-nguyen** [quay/quay#2716] [Q&A]
Hello, I'm testing Quay + Clair for a local deployment, and using S3 as an image storage backend. As I look in the docs, no backend configuration supports specifying the region of the bucket, both in [Project Quay](https://docs.projectquay.io/config_quay.html#config-fields-storage) and [Red Hat Quay](https://access.redhat.com/documentation/en-us/red_hat_quay/3.10/html/configure_red_hat_quay/config-fields-intro#config-fields-storage).

When Clair is pulling image layers from Quay, it uses region `us-east-1`, causing issues when bucket is in a different region.

Is it possible to specify bucket region? If so, how can I configure it?

Thanks!

**Comment by ibazulic** [comment]
Hi,

Yes, add the `s3_region: your_region` into the Quay config S3 driver parameters. That should fix the issue with downloading.

**Reply by tk-nguyen** [reply]
Thanks, it works great!

Another related question: If I were to use S3-compatible object storage (Wasabi, Backblaze, ...), how would I configure the region if I'm using `OCS/NooBaa` or `Ceph/RadosGW` backend?

**Reply by codespearhead** [reply]
Ceph Storage is S3-compatible, so you just need to change the env vars and [set a region that makes sense in your setup](https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/1.3/html/object_gateway_guide_for_red_hat_enterprise_linux/object_gateway_configuration_reference).

As for the other project, which I couldn't even find with Google, but the statement above still holds.
